src/server/analytic_services/prometheus_reports/noobaa_endpoint_report.js

```diff
@@ -192,6 +225,49 @@ class NooBaaEndpointReport extends BasePrometheusReport {
         metric.prom_instance.inc(labels, value);
     }
 
+    // Set value metric
+    set(name, labels, value) {
+        if (!this._metrics) return;
+        const metric = this._metrics[name];
+        if (!metric) throw new Error(`Unknown metric ${name}`);
+        if (metric.type !== 'Gauge') throw new Error(`Metric ${name} is not Gauge`);
+        metric.prom_instance.set(labels, value);
+    }
+
+    // setMultipleMetrics value metric
+    setMultipleMetrics(name, labels, values) {
```

Comment:
@naveenpaul1, as discussed, you can simplify this function by only maintaining the arrays of reports per label here and having a static collect method that averages the reports.